v2/parser/parse.go

```diff
@@ -143,6 +151,13 @@ func (p *Parser) LoadPackages(patterns ...string) error {
 	return err
 }
 
+// LoadPackagesWithConfig loads and parses the specified Go packages with the
+// specified packages.Config as a starting point.  This is for testing.
```

Comment:
"This is for testing" seems unlikely to hold if this allows injecting tricky things that impact the package load...

The surface area of the injected cfg is large and definitely allows stuff that would break us...
* some of which we stomp (Tests=false, Fset = ...)
* some of which we aggregate in ways I'm not sure will work (appending possibly more build tags, combining additional things with mode)
* some of which we don't do anything with in ways I'm not sure will work (allowing esoteric options like ParseFile / Overlay to rule)

Is there a way to limit the surface area of what we can inject this way to *just* things we know won't conflict with or break our internal use (e.g. Dir / Env)?

```go
if cfg != nil {
  cfgCopy := *cfg
  cfgCopy.Dir = ""
  cfgCopy.Env = nil
  if cfgCopy != (Config{}) {
    return fmt.Errorf("only Dir and Env may be set in the base config, got additional fields %#v", cfgCopy)
  }
}
```

Reply:
I could just take Dir and Env as args, or as you suggest above just extract those 2 fields from Config